### PR TITLE
Add changelog to documentation

### LINKED ISSUE #516: Include changelog in generated documentation
The generated documentation should include the changelog.

When we do this, we should also add a 'changelog' link in the project.urls in `pyproject.toml`.

### PR DESCRIPTION
**Pull request by mdickinson**
This PR adds the changelog to the Sphinx documentation, under the heading "Release history".

I also had to make a fix to the Sphinx configuration file in order to make the docs build at all, which suggest that we should have a CI build for the documentation. I'll make a separate issue for that.

Partial screenshot below. It's not pretty, but it's functional.

Closes #516

Screenshot 2023-03-22 at 15 43 12

### CODE REVIEW

**Comment by mdickinson** [comment]
> I'll make a separate issue for that.

See #553 (with fix in #552).

File: docs/source/index.rst [hunk 1]
```diff
@@ -7,9 +7,11 @@ Envisage Documentation
 
     envisage_core_documentation/index
     tasks_user_manual/index
+    changelog
```
Comment (corranwebster) [inline docs/source/index.rst:10]: Bikeshedding, but I'd be tempted to place this after the API docs, given that it is a fairly large chunk of the ToC and I think more people are likely to want to access the API docs than the changelog.

Comment (mdickinson) [reply docs/source/index.rst]: Done in 439a1cfacc71d6e877bcd988d21b9243da9ec412

**Comment by mdickinson** [comment]
Merged main in so that we can get the benefit of the doc build test workflow.

**Comment by mdickinson** [comment]
Doc build succeeds, and resulting docs are as expected.

**Pull request merged**